plugins/cache/redis.py

```diff
@@ -17,10 +17,15 @@
     options:
       _uri:
         description:
-          - A colon separated string of connection information for Redis.
-          - The format is V(host:port:db:password), for example V(localhost:6379:0:changeme).
+          - Redis connection string, 3 possible formats.
+          - The simple format consist in a colon separated string of connection information for Redis.
+          - This first format is V(host:port:db:password), for example V(localhost:6379:0:changeme).
           - To use encryption in transit, prefix the connection with V(tls://), as in V(tls://localhost:6379:0:changeme).
-          - To use redis sentinel, use separator V(;), for example V(localhost:26379;localhost:26379;0:changeme). Requires redis>=2.9.0.
+          - To use redis sentinel, use second format with separator V(;), for example V(localhost:26379;localhost:26379;0:changeme). Requires redis>=2.9.0.
+          - To specify advanced connection parameters (keep alive, etc ...), use the 3rd format.
+          - This last format consists in a '&' separated string of redis connection parameters as <name>=<value> pairs.
+          - For example V(host=localhost&port=6379&db=0&password=changeme&socket_keepalive=true).
+          - See https://redis-py2.readthedocs.io/en/latest/ for Redis connection parameters name and accepted values.
```

Comment:
Why not simply expose these extra parameters as proper options? That makes everything *a lot* more readable and thus simpler to use as well.

Reply:
Hello @felixfontein, i've suggested this because there are a lot of parameters in the Redis connection, and some complex ones like `socket_keepalive_options`. And I don't wanted to add too much complexity to the module, for some options not that much commonly used. But ok, maybe you're right, it might be better to expose all parameters as options. I'll prepare this, and I'll push the updated code here in few hours/days. Thank you !

Reply:
@pierre-claranet if I may suggest, one possible way would be to have a parameter (either a simple string or a list) for the user to pass those extra parameters. If that option is used, then it gets `'&'.join()`-ed with the URL.

Reply:
Another thing to keep in mind is that if `password=changeme` is used, as in the example, then this option should be marked `no_log: true`, personally I find that an undesirable side-effect, but YMMV.

Reply:
One thing to consider is also to consolidate the cache plugin's options with the module and/or lookup. Right now we have three separate pieces of code and options for handling Redis logins:
- The cache plugin (what this PR touches);
- The lookup plugin;
- The modules, which use the `redis` docs fragment and the `redis` module utils.

Reply:
Hello,

PR's code updated.

@russoz, thanks for your suggestions. I took care of obfuscating the Redis connection password in the verbose (-vvv) output. The password's value will be replaced with `****`. About your 1st suggestion, if I'm right, it's close to this PR's original code. The only difference would be to use an extra option, and not the `_url` one, to pass the extra parameters. And as discussed with @felixfontein previously, exposing options separately seems to be better for simplicity and readability.

@felixfontein, about the options consolidation, I'm not sure to completely understand your point. Do you suggest we use the same names, allowed values and format, for the same parameters in all pieces of codes ? For some parameters, for example these in Redis cache plugin `_uri` connection string (the host, port, db, password), I think we can't change them. The cache plugin documentation (https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#cache-plugins) says `New cache plugins should take the options _uri [...]`. Could you please add some precision, maybe one example, for you point ? Thanks.